tests/e2e/login_test.go

```diff
@@ -0,0 +1,97 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("odoLoginE2e", func() {
+	// user related constants
+	const loginTestUser = "testdeveloper"
+	const loginTestUserPassword = "testdeveloper"
+	const odoTestProject1 = "testproject1"
+
+	// Comand related constants
+	const baseOdoLoginCommand = "odo login"
+	const baseOdoProjectCreate = "odo project create"
+	const baseOdoProjectDelete = "odo project delete -f "
+	const ocWhoamiCommand = "oc whoami"
+	const ocTokenCommand = "oc whoami -t"
+
+	// variables to be used in test
+	var session string
+	var backToCurrentUserCommand string
+	var testUserLoginCommand string
+	var testUserLoginCommandWithToken string
+	var testUserLoginFailCommandWithToken string
+	var testUserCreateProject1Command string
+
+	Describe("Check for successful login and logout", func() {
+		Context("Initialize", func() {
+			It("Should initialize some variables", func() {
+				// Save currently logged in users token, so we can get back to that context after being done
+				t := runCmd(ocTokenCommand)
+				backToCurrentUserCommand = fmt.Sprintf("%s -t %s", baseOdoLoginCommand, t)
+				testUserLoginCommand = fmt.Sprintf("%s -u %s -p %s", baseOdoLoginCommand, loginTestUser, loginTestUserPassword)
+				testUserCreateProject1Command = fmt.Sprintf("%s %s", baseOdoProjectCreate, odoTestProject1)
+				testUserLoginFailCommandWithToken = fmt.Sprintf("%s -t verybadtoken", baseOdoLoginCommand)
+			})
+		})
+
+		Context("Run login tests", func() {
+			AfterEach(func() {
+				t := runCmd("oc projects -q")
+				projects := strings.Split(t, "\n")
+				for _, p := range projects {
+					if len(p) > 0 {
+						runCmd(fmt.Sprintf("%s %s", baseOdoProjectDelete, p))
+						time.Sleep(100 * time.Millisecond)
```

Comment:
Please do not use hard-coded waits may be you can try something similar to https://github.com/redhat-developer/odo/blob/master/tests/e2e/e2e.go#L51